homeassistant/components/fan/__init__.py

```diff
@@ -180,45 +193,57 @@ def handle_fan_service(service: str) -> None:
     return True
 
 
-class FanEntity(Entity):
+class FanEntity(ToggleEntity):
     """Representation of a fan."""
 
     # pylint: disable=no-self-use, abstract-method
 
-    def set_speed(self: Entity, speed: str) -> None:
+    @property
+    def state(self: ToggleEntity) -> str:
+        """Get the state of the fan."""
+        return self.__getattribute__(ATTR_SPEED)
+
+    def set_speed(self: ToggleEntity, speed: str) -> None:
         """Set the speed of the fan."""
         pass
 
-    def turn_on(self: Entity, speed: str=None) -> None:
+    def turn_on(self: ToggleEntity, speed: str=None, **kwargs) -> None:
         """Turn on the fan."""
-        pass
+        raise NotImplementedError()
 
-    def turn_off(self: Entity) -> None:
+    def turn_off(self: ToggleEntity, **kwargs) -> None:
         """Turn off the fan."""
-        pass
+        raise NotImplementedError()
 
-    def oscillate(self: Entity) -> None:
+    def oscillate(self: ToggleEntity, oscillating: bool) -> None:
         """Oscillate the fan."""
         pass
 
+    def is_on(self):
+        """Return true if the entity is on."""
+        return self.state not in [STATE_OFF, STATE_UNKNOWN]
```

Comment:
Implementing this method here and not in the platform makes it harder to check that a platform is correctly implemented. Now the platform has to implement the speed attribute and I think that is not as clearly defined in the code. It also restricts platforms as some fans might only have on/off.

I think you should consider leaving state property as in the parent ToggleEntity class and make `is_on` raise not implemented if not done in the platform. Make speed attribute a regular state attribute, similar to how a light with brightness works.